specifications/diagrams/state-diagrams/channel-state.txt

```diff
@@ -0,0 +1,12 @@
+stateDiagram-v2
+    [*] --> Waiting: NewChannel()
+    note left of [*]: Initiator opening a channel
+
+    Waiting --> Active: channel.Open()
+
+    Active --> Closing: channel.CloseStart()
+    
+    Closing --> Closed: Responder calls <br>channel.CloseCoordinated()
+    Closing --> Closed: channel.CloseUnCoordinated() 
```

Comment:
At a high-level I think this is right. I think the failure cases of the channel though are interesting and I wonder if we should capture them here or in another diagram. Most steps, if agreement isn't found, the participants should close the channel.